--- ./configure.ac.am112	2012-05-25 09:13:08.909839526 +0000
+++ ./configure.ac	2012-05-25 09:13:20.467839705 +0000
@@ -8,6 +8,8 @@ AM_CONFIG_HEADER([config.h])
 AC_CONFIG_AUX_DIR([build-aux])
 
 AM_INIT_AUTOMAKE([foreign dist-bzip2 -Wall -Werror])
+m4_pattern_allow([AM_PROG_AR])
+AM_PROG_AR
 LT_INIT
 AM_MAINTAINER_MODE
 
--- ./spice-common/configure.ac.am112	2012-05-25 09:13:29.533838518 +0000
+++ ./spice-common/configure.ac	2012-05-25 09:13:39.213838740 +0000
@@ -13,6 +13,8 @@ AC_CONFIG_AUX_DIR([build-aux])
 AM_INIT_AUTOMAKE([1.11 dist-xz no-dist-gzip tar-ustar foreign -Wall -Werror])
 AM_MAINTAINER_MODE
 AM_SILENT_RULES([yes])
+m4_pattern_allow([AM_PROG_AR])
+AM_PROG_AR
 LT_INIT
 
 AC_PROG_CC
