diff -up spice-gtk-0.15/configure.ac.automake-1_13 spice-gtk-0.15/configure.ac
--- spice-gtk-0.15/configure.ac.automake-1_13	2012-12-20 21:13:30.000000000 +0100
+++ spice-gtk-0.15/configure.ac	2013-01-02 16:13:14.471737538 +0100
@@ -4,7 +4,7 @@ AC_INIT([spice-gtk], [m4_esyscmd(build-a
         [spice-devel@lists.freedesktop.org])
 
 AC_CONFIG_MACRO_DIR([m4])
-AM_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_AUX_DIR([build-aux])
 
 AM_INIT_AUTOMAKE([foreign dist-bzip2 -Wall -Werror])
@@ -71,8 +71,11 @@ AC_CHECK_HEADERS([termios.h])
 AC_CHECK_LIBM
 AC_SUBST(LIBM)
 
+PKG_CHECK_MODULES(PROTOCOL, spice-protocol >= 0.12.2)
+AC_SUBST(PROTOCOL_CFLAGS)
+
 AC_CONFIG_SUBDIRS([spice-common])
-COMMON_CFLAGS='-I ${top_srcdir}/spice-common/ -I ${top_srcdir}/spice-common/spice-protocol/'
+COMMON_CFLAGS='-I ${top_srcdir}/spice-common/ ${PROTOCOL_CFLAGS}'
 AC_SUBST(COMMON_CFLAGS)
 
 dnl =========================================================================
diff -up spice-gtk-0.15/spice-common/common/Makefile.am.automake-1_13 spice-gtk-0.15/spice-common/common/Makefile.am
--- spice-gtk-0.15/spice-common/common/Makefile.am.automake-1_13	2012-11-07 14:29:26.000000000 +0100
+++ spice-gtk-0.15/spice-common/common/Makefile.am	2013-01-02 16:12:19.699456192 +0100
@@ -14,7 +14,7 @@ SERVER_MARSHALLERS =					\
 	$(srcdir)/generated_server_marshallers.h	\
 	$(NULL)
 
-BUILT_SOURCES = $(CLIENT_MARSHALLERS) $(SERVER_MARSHALLERS) $(top_srcdir)/spice-protocol/spice/enums.h
+BUILT_SOURCES = $(CLIENT_MARSHALLERS) $(SERVER_MARSHALLERS)
 
 noinst_LTLIBRARIES = libspice-common.la libspice-common-server.la libspice-common-client.la
 libspice_common_la_SOURCES =		\
@@ -121,12 +121,6 @@ generated_server_marshallers.c: $(top_sr
 generated_server_marshallers.h: $(top_srcdir)/spice.proto $(MARSHALLERS_DEPS)
 	$(AM_V_GEN)$(PYTHON) $(top_srcdir)/spice_codegen.py --generate-marshallers $(STRUCTS) --server --include messages.h -H $< $@ >/dev/null
 
-# this is going to upset automake distcheck, since we try to write to
-# readonly srcdir. To limit the fail chances, rebuild automatically
-# enums.h only if the spice.proto has changed.
-$(top_srcdir)/spice-protocol/spice/enums.h: $(top_srcdir)/spice.proto # $(MARSHALLERS_DEPS)
-	$(AM_V_GEN)$(PYTHON) $(top_srcdir)/spice_codegen.py --generate-enums $< $@ >/dev/null
-
 EXTRA_DIST =				\
 	$(CLIENT_MARSHALLERS)		\
 	$(SERVER_MARSHALLERS)		\
diff -up spice-gtk-0.15/spice-common/configure.ac.automake-1_13 spice-gtk-0.15/spice-common/configure.ac
--- spice-gtk-0.15/spice-common/configure.ac.automake-1_13	2012-11-07 14:29:26.000000000 +0100
+++ spice-gtk-0.15/spice-common/configure.ac	2013-01-02 16:12:19.713456008 +0100
@@ -6,7 +6,7 @@ AC_INIT([spice-common],
 
 AC_CONFIG_SRCDIR([common/bitops.h])
 AC_CONFIG_MACRO_DIR([m4])
-AM_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_AUX_DIR([build-aux])
 
 # For automake >= 1.12
@@ -26,8 +26,7 @@ fi
 AM_PROG_CC_C_O
 
 # Checks for libraries
-AC_CONFIG_SUBDIRS([spice-protocol])
-PROTOCOL_CFLAGS='-I ${top_srcdir}/spice-protocol'
+PKG_CHECK_MODULES(PROTOCOL, spice-protocol >= 0.12.2)
 AC_SUBST(PROTOCOL_CFLAGS)
 
 PKG_CHECK_MODULES(PIXMAN, pixman-1 >= 0.17.7)
diff -up spice-gtk-0.15/spice-common/Makefile.am.automake-1_13 spice-gtk-0.15/spice-common/Makefile.am
--- spice-gtk-0.15/spice-common/Makefile.am.automake-1_13	2012-11-07 14:29:26.000000000 +0100
+++ spice-gtk-0.15/spice-common/Makefile.am	2013-01-02 16:12:19.726455836 +0100
@@ -2,7 +2,7 @@ NULL =
 ACLOCAL_AMFLAGS = -I m4
 
 SUBDIRS = python_modules common
-DIST_SUBDIRS = spice-protocol $(SUBDIRS)
+DIST_SUBDIRS = $(SUBDIRS)
 
 EXTRA_DIST =				\
 	spice_codegen.py		\

